### DISCUSSION TITLE
An SDK for logto-management-api

### DISCUSSION THREAD

**Discussion by IRediTOTO** [logto-io/logto#6608] [Q&A]
Hello, I am trying to use machine to machine app, I see this api docs [https://bump.sh/logto/doc/logto-management-api/operation/operation-createuser](https://bump.sh/logto/doc/logto-management-api/operation/operation-createuser)
But Do you have a package like "@logto-management" ? Or do I must write function to call api one by one?

**Comment by xiaoyijun** [comment]
Sorry for the late reply!

Yes, for JavaScript and TypeScript there is now an official package: [`@logto/api`](https://www.npmjs.com/package/@logto/api). It provides Management API types and clients generated from the OpenAPI spec, with built-in client credentials authentication, so you do not need to write the calls one by one.

For other languages, you can generate a typed client from the [OpenAPI spec](https://openapi.logto.io/) with tools like `openapi-generator`.

Either way, create a machine-to-machine app and fetch access tokens via client credentials. See [Interact with Management API](https://docs.logto.io/integrate-logto/interact-with-management-api) for a step-by-step guide.
